### PR TITLE
cranelift: Rework pinned register lowering

### PR DESCRIPTION
**Pull request by elliottt**
Rework pinned register lowering to avoid the use of pinned virtual registers, instead using the `MovFromPReg` and `MovToPReg` pseudo instructions.

### CODE REVIEW

File: cranelift/codegen/src/isa/aarch64/inst.isle [hunk 1]
```diff
@@ -167,10 +167,16 @@
 
        ;; Like `Move` but with a particular `PReg` source (for implementing CLIF
        ;; instructions like `get_stack_pointer`).
-       (MovPReg
+       (MovFromPReg
         (rd WritableReg)
         (rm PReg))
 
+       ;; Like `Move` but with a particular `PReg` destination (for
+       ;; implementing CLIF instructions like `get_stack_pointer`).
```
Comment (fitzgen) [inline cranelift/codegen/src/isa/aarch64/inst.isle:175]: These docs are a copy paste that should be updated for the actual use of this instruction.

File: cranelift/filetests/filetests/isa/x64/pinned-reg.clif [hunk 2]
```diff
@@ -28,8 +30,14 @@ block0:
 
 ;   pushq   %rbp
 ;   movq    %rsp, %rbp
+;   subq    %rsp, $16, %rsp
+;   movq    %rsi, 0(%rsp)
```
Comment (elliottt) [inline cranelift/filetests/filetests/isa/x64/pinned-reg.clif:34]: It's not clear to me why this test changed to use `%rsi` for the temporary, but that's what causes the new stack space.

File: cranelift/codegen/src/isa/x64/inst.isle [hunk 3]
```diff
@@ -95,8 +95,13 @@
 
        ;; Like `MovRR` but with a physical register source (for implementing
        ;; CLIF instructions like `get_stack_pointer`).
-       (MovPReg (src PReg)
-                (dst WritableGpr))
+       (MovFromPReg (src PReg)
+                    (dst WritableGpr))
+
+       ;; Like `MovRR` but with a physical register destination (for
+       ;; implementing CLIF instructions like `get_pinned_reg`).
```
Comment (fitzgen) [inline cranelift/codegen/src/isa/x64/inst.isle:102]: Ditto

**Review by fitzgen — APPROVED** [review]
LGTM with nitpicks below addressed

File: cranelift/filetests/filetests/isa/x64/pinned-reg.clif (hunk 2 above)
Comment (elliottt) [reply cranelift/filetests/filetests/isa/x64/pinned-reg.clif:34]: Scratch that -- it's the first preferred register, and also a clobber with sysv64.

File: cranelift/codegen/src/isa/aarch64/inst.isle (hunk 1 above)
Comment (elliottt) [reply cranelift/codegen/src/isa/aarch64/inst.isle]: Thanks for catching that!

File: cranelift/codegen/src/isa/x64/inst.isle (hunk 3 above)
Comment (elliottt) [reply cranelift/codegen/src/isa/x64/inst.isle]: Thanks for catching that!

**Pull request merged**